src/service/xhr-impl.js

```diff
@@ -213,7 +213,15 @@ export class Xhr {
    */
   fetchJson(input, opt_init, opt_allowFailure) {
     const init = setupInit(opt_init, 'application/json');
+    const headerContentType = init.headers['Content-Type'];
     if (init.method == 'POST' && !isFormData(init.body)) {
+      const isRequestBodyFormUrlEncoded =
+        (headerContentType ===
+          'application/x-www-form-urlencoded;charset=utf-8');
+      if (isRequestBodyFormUrlEncoded &&
+        isString(init.body)) {
```

Comment:
nit: indent

But I'd say that just testing the `content-type` should be sufficient.

Reply:
done